go.mod

```diff
@@ -11,7 +11,7 @@ require (
 	github.com/aws/aws-sdk-go v1.35.3
 	github.com/benbjohnson/clock v1.1.0
 	github.com/bradleyjkemp/grpc-tools v0.2.5
-	github.com/cenkalti/backoff v2.2.1+incompatible
+	github.com/cenkalti/backoff/v4 v4.0.2
```

Comment:
Currently, the retry behavior and backoff behavior in `pkg/retry` is tightly-coupled with each other. It can't be conveniently used in this scenario. Maybe we can decouple them later and have a better implementation.